python/lsst/meas/algorithms/skyObjects.py

```diff
@@ -33,9 +35,11 @@ def generateSkyObjects(mask, seed, config):
     through the provided `mask` (in which objects are typically flagged
     as `DETECTED`).
 
-    The algorithm for determining sky objects is random trial and error:
-    we try up to `nTrialSkySources` random positions to find `nSources`
-    sky objects.
+    Sky objects are positioned using a quasi-random Halton sequence number
+    generator. This is a deterministic sequence that mimics a random trial and
+    error approach whilst acting to minimize clustering of points for a given
+    field of view. Up to `nTrialSources` points are generated, returning the
+    first `nSources` that do not overlap with the mask.
```

Comment:
RE: 
> "acting to minimize clustering of points"

does this algorithm guarantee no overlapping positions?  If not, you might consider adding:

```
# Add doubled-in-size sky object spanSet to the avoid mask.
avoid = avoid.union(spans.dilated(int(skySourceRadius)))
```
after the line:

```
skyFootprints.append(fp)
```
at line 93  below in order to have a cumulative `avoid` mask.  It worked nicely using the old trial 'n error algorithm (but may not be necessary now?)

Reply:
There is nothing explicit in the SciPy Halton Sequence implementation that prevents overlapping positions as N tends to infinity, however, it is (by design) exceptionally unlikely.

However, I do like your additional logic to prevent this and I don't think it adds too much overhead. As `SkyObjectsConfig` already has a `growMask` field, I'm tempted to recommend using this information rather than hard-coding a double-in-size condition?

Reply:
Following testing, I think I convinced myself that I prefer your initial idea of growing the sky object footprints by their own radius. I experimented with growing the `growMask` parameter, and while it works, it occurred to me that that parameter was intended to mask the flux in the wings of astrophysical sources. We want to grow sky object footprints for a different reason altogether (to avoid clustering for stats purposes). To that end, I've adopted your initial recommendation above as-is.